grails-app/services/org/pih/warehouse/invoice/PrepaymentInvoiceService.groovy

```diff
@@ -284,7 +288,11 @@ class PrepaymentInvoiceService {
         }
 
         items.each { Map item ->
```

Comment:
I know this isn't new here, but I generally prefer to strongly type Lists and then do `for (InvoiceItem item : items)` loops instead of closures to avoid the extra overhead involved with closures. Stack traces also render more properly in regular loops and we can also do break/continue more easily in loops.

Reply:
I agree with typing of variables whenever possible, I've raised that topic many times not to throw `def` when it's not a dynamic variable, but I don't have much knowledge what the `items` are. I can see that type `List` is passed, so whether those are some kind of items that come from the frontend and are difficult to parse or ....